### ISSUE TITLE
Can't resolve 'react-dnd/lib/DragDropContext' with react-dnd@7.1.0

### ISSUE THREAD

**Issue by imsobear** [frontend-collective/react-sortable-tree#463]
# Reporting a Bug?

Can't resolve 'react-dnd/lib/DragDropContext' with react-dnd@7.1.0

![image](https://user-images.githubusercontent.com/2505411/53946067-ec7c8d80-40fd-11e9-9045-9a7a626c58e1.png)

react-sortable-tree@2.6.0 code:

![image](https://user-images.githubusercontent.com/2505411/53946104-0322e480-40fe-11e9-9494-6ec7f5794ad9.png)

react-dnd@7.1.0 looks like changed `lib/` directory tree. https://github.com/react-dnd/react-dnd/pull/1205:

![image](https://user-images.githubusercontent.com/2505411/53946238-56953280-40fe-11e9-9162-7f97b7003a93.png)

react-sortable-tree@2.6.0 package.json，will auto update react-dnd@7.x:

```
"dependencies": {
    "frontend-collective-react-dnd-scrollzone": "^1.0.1",
    "lodash.isequal": "^4.5.0",
    "prop-types": "^15.6.1",
    "react-dnd": "^7.0.1",
    "react-dnd-html5-backend": "^7.0.1",
    "react-lifecycles-compat": "^3.0.4",
    "react-virtualized": "^9.19.1"
}
```

**Comment by imsobear** [comment]
https://github.com/react-dnd/react-dnd/pull/1205/files

**Comment by intizarahmad** [comment]
We are also facing the same issue, when can we expect the merge of this PR?

**Comment by intizarahmad** [comment]
We are importing 
import { Consumer as DragDropContextConsumer } from 'react-dnd/lib/DragDropContext';
 but react-dnd has changed their lib path from react-dnd/lib/DragDropContext to react-dnd/lib/esm/DragDropContext 
Please fix this
![image](https://user-images.githubusercontent.com/2504495/53978967-83fbd380-4132-11e9-8c4e-ccbfbb18a2ac.png)

**Comment by quigleyj-mavenomics** [comment]
If you're using this package in a top-level project, you can use a mechanism like [Yarn `resolutions`](https://yarnpkg.com/lang/en/docs/selective-version-resolutions/) or [`npm-force-resolutions`](https://github.com/rogeriochaves/npm-force-resolutions) to override the resolved version of `react-dnd` and force a good version (like `7.0.2`).

This did not work for my use case, so I've published a forked version of this package as a temporary workaround until this issue is fixed: [`fixed-react-sortable-tree`](https://www.npmjs.com/package/fixed-react-sortable-tree)

**Comment by zakkudo** [comment]
It would be great if this was fixed.

**Comment by intizarahmad** [comment]
https://github.com/react-dnd/react-dnd/issues/1247 
See the comment from react-dnd:
It should be in react-dnd/lib/cjs/DragDropContext.js
With the 7.1 release, we added separate builds for ES Modules and CommonJS. Previously they had all been CommonJS. I assumed that most people were importing the types at the library's top level, so I made this a minor version, e.g.:

import {DragDropContext} from 'react-dnd'

**Comment by dolezel** [comment]
@intizarahmad We are not using `DragDropContext` but `Consumer`, which is not exported.

**Comment by ecnaidar** [comment]
Someone probably also should get issue going in `react-dnd` repository to export `DragDropConsumer` so this issue could be avoided in future. 
Was not very fun to suddenly not be able to build a project without changing anything.
For yarn adding to 
```
 "resolutions": {
    "react-sortable-tree/**/react-dnd": "7.0.2"
  }
```
to `package.json` did the trick, thanks @quigleyj-mavenomics for pointing this out

**Comment by dolezel** [comment]
@wuweiweiwu Will you have time to look on it and eventually publish new version?

**Comment by sunndeath** [comment]
Guys, where is a solution?

**Comment by ecnaidar** [comment]
@sunndeath 
There are multiple instructions how to immediately unfuck the build process in this thread.
Fix seems to be merged but is not released yet.

**Comment by sunndeath** [comment]
> @sunndeath
> There are multiple instructions how to immediately unfuck the build process in this thread.
> Fix seems to be merged but is not released yet.

refactor from:
`import { DragDropContext } from 'react-dnd';`
to:
`import DragDropContext from 'react-dnd/lib/cjs/DragDropContext.js';`

don't work form me,

where is the another way to fix it?

**Comment by ecnaidar** [comment]
Wonder where you tried to put that... 
It is a solution to fix library itself not how you use it.

https://github.com/frontend-collective/react-sortable-tree/issues/463#issuecomment-470654505

**Comment by Pavel910** [comment]
This is the reason `semver` exists and MUST be respected! You don't just publish a major breaking change under a `minor` release version 😠 

Thanks @quigleyj-mavenomics and @ecnaidar for the temporary solution.

**Comment by Alexixyc** [comment]
como on~please fix it~~~😭

**Comment by Alexixyc** [comment]
come on~please fix it~~~😭

**Comment by wuweiweiwu** [comment]
released in 2.6.1!!! Thanks for waiting!
